rpcserver.go

```diff
@@ -4248,6 +4248,40 @@ func (r *rpcServer) ListPayments(ctx context.Context,
 			return nil, err
 		}
 
+		htlcs := make([]*lnrpc.HTLCAttempt, 0, len(payment.HTLCs))
+		for _, htlc := range payment.HTLCs {
+			var (
+				status      lnrpc.HTLCAttempt_HTLCStatus
+				resolveTime time.Time
+			)
+			switch {
+			case htlc.Settle != nil:
+				status = lnrpc.HTLCAttempt_SUCCEEDED
+				resolveTime = htlc.Settle.SettleTime
+
+			case htlc.Failure != nil:
+				status = lnrpc.HTLCAttempt_FAILED
+				resolveTime = htlc.Failure.FailTime
+
+			default:
+				status = lnrpc.HTLCAttempt_IN_FLIGHT
+			}
+
+			htlcPath := make([]string, len(htlc.Route.Hops))
+			for i, hop := range htlc.Route.Hops {
+				htlcPath[i] = hex.EncodeToString(
+					hop.PubKeyBytes[:],
+				)
```

Comment:
Why not `repeated bytes` and format it in `lncli`?

Reply:
now uses bytes and formatted in `lncli`. the changes spread quite a lot tho as you'll see. maybe it's time we fork `jsonpb` to do this for us instead of needing to manually do this workaround